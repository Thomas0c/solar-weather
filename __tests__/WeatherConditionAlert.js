import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import WeatherConditionAlert from '../lib/js/app/components/weatherConditionAlert';

test('renders correctly when showAlert is true', () => {
  const tree = renderer.create(<WeatherConditionAlert summary="Test Summary" showAlert={1} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly when showAlert is false', () => {
  const tree = renderer.create(<WeatherConditionAlert summary="Test Summary" showAlert={0} />).toJSON();
  expect(tree).toMatchSnapshot();
});
