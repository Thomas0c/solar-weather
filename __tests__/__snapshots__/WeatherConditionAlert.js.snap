// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when showAlert is false 1`] = `
<Text
  accessible={true}
  adjustsFontSizeToFit={undefined}
  allowFontScaling={true}
  ellipsizeMode="tail"
  minimumFontScale={undefined}
  numberOfLines={undefined}
  onLayout={undefined}
  onLongPress={undefined}
  onPress={undefined}
  pressRetentionOffset={undefined}
  selectable={undefined}
  selectionColor={undefined}
  style={
    Array [
      Object {
        "color": "#F9F9F9",
        "fontFamily": "HelveticaNeue",
        "fontSize": 24,
        "fontWeight": "600",
        "marginBottom": 20,
      },
      Object {
        "color": "#EFEFEF",
      },
    ]
  }
  suppressHighlighting={undefined}
  testID={undefined}
  textBreakStrategy={undefined}
>
  Test Summary
</Text>
`;

exports[`renders correctly when showAlert is true 1`] = `
<Text
  accessible={true}
  adjustsFontSizeToFit={undefined}
  allowFontScaling={true}
  ellipsizeMode="tail"
  minimumFontScale={undefined}
  numberOfLines={undefined}
  onLayout={undefined}
  onLongPress={undefined}
  onPress={undefined}
  pressRetentionOffset={undefined}
  selectable={undefined}
  selectionColor={undefined}
  style={
    Array [
      Object {
        "color": "#F9F9F9",
        "fontFamily": "HelveticaNeue",
        "fontSize": 24,
        "fontWeight": "600",
        "marginBottom": 20,
      },
      Object {
        "color": "#EFEFEF",
      },
    ]
  }
  suppressHighlighting={undefined}
  testID={undefined}
  textBreakStrategy={undefined}
>
  Test Summary
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "height": 25,
          "marginTop": 3,
          "width": 25,
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      accessibilityLabel={undefined}
      accessible={undefined}
      blurRadius={undefined}
      capInsets={undefined}
      defaultSource={undefined}
      onError={undefined}
      onLayout={undefined}
      onLoad={undefined}
      onLoadEnd={undefined}
      onLoadStart={undefined}
      onPartialLoad={undefined}
      onProgress={undefined}
      resizeMethod={undefined}
      resizeMode={undefined}
      source={
        Object {
          "testUri": "../../../assets/alert_icon.png",
        }
      }
      style={
        Object {
          "height": 20,
          "marginLeft": 10,
          "marginTop": 5,
          "resizeMode": "contain",
          "width": 20,
        }
      }
      testID={undefined}
    />
  </View>
</Text>
`;
